packages/astro/src/core/config/config.ts

```diff
@@ -157,82 +155,74 @@ interface LoadConfigOptions {
  */
 export async function resolveConfigPath(
-	configOptions: ResolveConfigPathOptions
+	options: ResolveConfigPathOptions
 ): Promise<string | undefined> {
-	const root = resolveRoot(configOptions.cwd);
-	const flags = resolveFlags(configOptions.flags || {});
-
 	let userConfigPath: string | undefined;
-	if (flags?.config) {
-		userConfigPath = /^\.*\//.test(flags.config) ? flags.config : `./${flags.config}`;
-		userConfigPath = fileURLToPath(new URL(userConfigPath, `file://${root}/`));
-		if (!configOptions.fs.existsSync(userConfigPath)) {
+	if (options.configFile) {
+		userConfigPath = path.join(options.root, options.configFile);
+		if (!options.fs.existsSync(userConfigPath)) {
 			throw new AstroError({
 				...AstroErrorData.ConfigNotFound,
-				message: AstroErrorData.ConfigNotFound.message(flags.config),
+				message: AstroErrorData.ConfigNotFound.message(options.configFile),
 			});
 		}
 	} else {
-		userConfigPath = await search(configOptions.fs, root);
+		userConfigPath = await search(options.fs, options.root);
 	}
 
 	return userConfigPath;
 }
 
-interface OpenConfigResult {
-	userConfig: AstroUserConfig;
-	astroConfig: AstroConfig;
-	flags: CLIFlags;
-	root: string;
-}
-
 /** Load a configuration file, returning both the userConfig and astroConfig */
-export async function openConfig(configOptions: LoadConfigOptions): Promise<OpenConfigResult> {
+// NOTE: This function will be removed in a later PR. Use `resolveConfig` instead.
+export async function openConfig(configOptions: LoadConfigOptions): Promise<ResolveConfigResult> {
 	const root = resolveRoot(configOptions.cwd);
 	const flags = resolveFlags(configOptions.flags || {});
 
-	const userConfig = await loadConfig(configOptions, root);
-	const astroConfig = await resolveConfig(userConfig, root, flags, configOptions.cmd);
+	const userConfig = await loadConfig(root, flags.config, configOptions.fsMod);
+	const astroConfig = await resolveConfigLegacy(userConfig, root, flags, configOptions.cmd);
 
 	return {
-		astroConfig,
 		userConfig,
-		flags,
-		root,
+		astroConfig,
 	};
 }
 
 async function loadConfig(
-	configOptions: LoadConfigOptions,
-	root: string
+	root: string,
+	configFile?: string | false,
+	fsMod = fs
 ): Promise<Record<string, any>> {
-	const fsMod = configOptions.fsMod ?? fs;
+	if (configFile === false) return {};
+
 	const configPath = await resolveConfigPath({
-		cwd: configOptions.cwd,
-		flags: configOptions.flags,
+		root,
+		configFile,
 		fs: fsMod,
 	});
 	if (!configPath) return {};
 
 	// Create a vite server to load the config
 	return await loadConfigWithVite({
+		root,
 		configPath,
 		fs: fsMod,
-		root,
 	});
 }
 
 /** Attempt to resolve an Astro configuration object. Normalize, validate, and return. */
-export async function resolveConfig(
+// NOTE: This function will be removed in a later PR. Renamed with "Legacy" suffix to make way
+// for the new `resolveConfig` API.
+async function resolveConfigLegacy(
```

Comment:
Un-export as it's not used internally. Also rename to make way for new API name.